### PR TITLE
Add bulk:deleteByQuery

### PR DESCRIPTION
**Pull request by Aschen**
## What does this PR do ?

Adds `bulk:deleteByQuery` method.

This method is more directly linked to ES original `delete_by_query` method.  
The purpose is to bypass limitations of `document:deleteByQuery` about the limit of deleted documents.

You may want to deletes millions of documents matching a query but since Kuzzle has to trigger realtime notifications for every method of the document controller, the `document:deleteByQuery` method has to fetch document before delete them. That's why there is a limit.

But if you only want to delete a lot of documents matching a query you will have to make calls to `document:search` and then `document:delete`. For 200 000 documents and the default limit (200 documents), you will have to make 1000 calls to both methods which is unacceptable

### CODE REVIEW

File: doc/2/api/controllers/document/delete-by-query/index.md [hunk 1]
```diff
@@ -14,7 +14,13 @@ Documents removed that way trigger real-time notifications.
 
 The request fails if the number of documents returned by the search query exceeds the `documentsWriteCount` server configuration (see the [Configuring Kuzzle](/core/2/guides/essentials/configuration) guide).
 
-To remove a greater number of documents, either change the server configuration, or split the search query.
+This behavior is to prevent Kuzzle from Out of Memory error or DOS because every deleted document has to be fetched first to trigger realtime notifications.
```
Comment (scottinet) [inline doc/2/api/controllers/document/delete-by-query/index.md:17]: I don't think using `deleteByQuery` can be considered a DoS attack.

```suggestion
This behavior aims at limiting the pressure on memory and on real-time notifications.
```

File: doc/2/api/controllers/document/delete-by-query/index.md [hunk 2]
```diff
@@ -61,6 +68,10 @@ Body:
 - `collection`: collection name
 - `index`: index name
 
+### Optional:
```
Comment (scottinet) [inline doc/2/api/controllers/document/delete-by-query/index.md:71]:
```suggestion
### Optional
```

File: doc/2/api/controllers/document/delete-by-query/index.md [hunk 3]
```diff
@@ -61,6 +68,10 @@ Body:
...
+
+- `refresh`: if set to `wait_for`, Kuzzle will not respond until the created/replaced documents are indexed
```
Comment (scottinet) [inline doc/2/api/controllers/document/delete-by-query/index.md:73]:
```suggestion
- `refresh`: if set to `wait_for`, Kuzzle will not respond until the deleted documents are removed from the search indexes
```

File: doc/2/api/controllers/bulk/delete-by-query/index.md [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+```
+
+---
+
+## Arguments
+
+- `collection`: collection name
+- `index`: index name
+
+### Optional:
```
Comment (scottinet) [inline doc/2/api/controllers/bulk/delete-by-query/index.md:59]:
```suggestion
### Optional
```

File: doc/2/api/controllers/bulk/delete-by-query/index.md [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+
+- `refresh`: if set to `wait_for`, Kuzzle will not respond until the deleted documents are indexed
```
Comment (scottinet) [inline doc/2/api/controllers/bulk/delete-by-query/index.md:61]:
```suggestion
- `refresh`: if set to `wait_for`, Kuzzle will not respond until the deleted documents are removed from the search indexes
```

File: test/api/controllers/bulk.test.js [hunk 6]
```diff
@@ -188,4 +188,34 @@ describe('Test the bulk controller', () => {
         request, mCreateOrReplaceResult, true);
     });
   });
+
+  describe.only('#deleteByQuery', async () => {
```
Comment (scottinet) [inline test/api/controllers/bulk.test.js:192]:
```suggestion
  describe('#deleteByQuery', async () => {
```

File: doc/2/api/controllers/document/delete-by-query/index.md (hunk 1 above)
Comment (Aschen) [reply doc/2/api/controllers/document/delete-by-query/index.md:17]: I meant that the OOM can lead to DOS but I agree it could be confusing

**Pull request merged**